src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,142 @@
+                }
+            }
+            return keys;
+        }
+    }
+
+    public boolean containsKey(K key) {
+        if (size == 0) {
+            return false;
+        }
+        int ifItWasFound = 0;
+        for (K currentKey : keySet()) {
+            if ((key != null && currentKey != null && currentKey.equals(key))
+                    || key == null && currentKey == null) {
+                ifItWasFound++;
+            }
+        }
+        return ifItWasFound != 0;
+    }
+
+    public V[] values() {
+        V[] values = (V[]) new Object[size];
+        return values;
+    }
+
+    private HashObject getHashObject(K key) {
+        HashObject currentNode = hashMap[keyIndex(key)];
+        while (!((currentNode.key != null && key != null && currentNode.key.equals(key))
+                || (currentNode.key == null && key == null))) {
+            currentNode = currentNode.next;
+        }
+        return currentNode;
+    }
 
     @Override
     public void put(K key, V value) {
-
+        if (!containsKey(key)) {
+            add(key, value);
+        } else {
+            set(key, value);
+        }
     }
 
     @Override
     public V getValue(K key) {
-        return null;
+        if (!containsKey(key)) {
+            return null;
+        } else {
+            return (V) getHashObject(key).value;
+        }
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private static class HashObject<K, V> {
+        private K key;
+        private V value;
+        private HashObject<K, V> next;
+
+        public HashObject(K key, V value, HashObject<K, V> next) {
+            this.key = key;
+            this.value = value;
+            this.next = next;
+        }
+    }
+
+    private void enlarge() {
+        K[] listOfKeys = (K[]) new Object[size];
+        V[] listOfValues = (V[]) new Object[size];
+        for (int i = 0; i < size; i++) {
+            listOfKeys[i] = keySet()[i];
+            listOfValues[i] = getValue(keySet()[i]);
+        }
+        hashMap = new HashObject[capacity * 2];
+        size = 0;
+        capacity = capacity * 2;
+
+        for (int i = 0; i < listOfKeys.length; i++) {
+            put(listOfKeys[i], listOfValues[i]);
+        }
+    }
+
+    private int keyIndex(K key) {
+        if (key == null) {
+            return 0;
+        } else {
+            return abs(key.hashCode()) % capacity;
+        }
+    }
+
+    private void add(K key, V value) {
+        if (size == (int) (capacity * DEFAULT_LOAD_FACTOR)) {
+            enlarge();
+        }
+        if (hashMap[keyIndex(key)] == null) {
+            hashMap[keyIndex(key)] = new HashObject<>(key, value, null);
+        } else {
+            HashObject currentNode = hashMap[keyIndex(key)];
+            while (currentNode.next != null) {
+                currentNode = currentNode.next;
+            }
+            currentNode.next = new HashObject<>(key, value, null);
```

Comment:
The `add` method could be more efficient. Instead of iterating through all nodes in the bucket to find the last node, you could directly add the new node at the beginning of the bucket. This would make the method run in constant time instead of linear time.